common/wuanalysis/anarule.cpp

```diff
@@ -165,11 +165,42 @@ class IoSkewRule : public AActivityRule
     const char * category;
 };
 
+class KeyedJoinExcessRejectedRowsRule : public ActivityKindRule
+{
+public:
+    KeyedJoinExcessRejectedRowsRule() : ActivityKindRule(TAKkeyedjoin) {}
+
+    virtual bool check(PerformanceIssue & result, IWuActivity & activity, const WuAnalyseOptions & options) override
+    {
+        stat_type preFiltered = activity.getStatRaw(StNumPreFiltered);
+        if (preFiltered)
+        {
+            IWuEdge * inputEdge = activity.queryInput(0);
+            stat_type rowscnt = inputEdge->getStatRaw(StNumRowsProcessed);
+            if (rowscnt)
+            {
+                stat_type preFilteredPer = statPercent( (long double) preFiltered * 100.0 / rowscnt );
```

Comment:
why long double?  double is fine.

Reply:
hmm I see the function takes a long double.  That is unusual.

Reply:
@ghalliday Shall I change it to double?

Reply:
@shamser I debated this for a while.  I was inclined to merge as-is, but since it is an overloaded function I think it would be better to change it to double (otherwise everything that wants to call it needs an unusual cast).  As far as I can see it was not previously called.